### PR TITLE
Theme related

### PR DESCRIPTION
**Pull request by KennethicEnergy**
This commit contains theme related changes, including dark and light mode and also swatches for header color.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
+			<Router>
+				<Navbar />
+				<ThemeSelector />
+				<Routes>
+					{routes.map((route) => (
+						<Route
+							key={route.id}
+							exact={route.isExact}
+							path={route.path}
+							element={route.component}></Route>
```
Comment (neil-borbon) [inline src/App.js:24]: Fix Code format of <Route>

File: src/components/recipe-list/Recipe-List.css [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 .recipe-list .card:hover {
   transform: rotate(3deg);
-  border: 1px solid #58249c;
+  border: 1px solid #555;
```
Comment (neil-borbon) [inline src/components/recipe-list/Recipe-List.css:21]: Use "rem"

File: src/components/recipe-list/Recipe-List.css [hunk 3]
```diff
@@ -49,4 +49,17 @@
...
+.recipe-list .card.dark {
+  background: #555;
+}
+.recipe-list .card.dark p,
+.recipe-list .card.dark h3,
+.recipe-list .card.dark div {
+  color: #e4e4e4;
+}
+.card.dark:hover {
+  border: 1px solid #ffffff;
```
Comment (neil-borbon) [inline src/components/recipe-list/Recipe-List.css:64]: Use "rem"

File: src/components/recipe-list/Recipe-List.js [hunk 4]
```diff
@@ -1,23 +1,24 @@
...
+	if (recipes.length === 0) {
+		return <div className={`error ${mode}`}>No recipes to load...</div>;
+	}
+	return (
+		<div className='recipe-list'>
+			{recipes.map((recipe) => (
+				<div className={`card ${mode}`} key={recipe.id}>
+					<h3>{recipe.title}</h3>
+					<p>{recipe.cookingTime} to make.</p>
+					<div>{recipe.method.substring(0, 100)}...</div>
```
Comment (neil-borbon) [inline src/components/recipe-list/Recipe-List.js:18]: Move this to a util folder and have a function return the value for this

File: src/components/searchbar/Searchbar.js [hunk 5]
```diff
@@ -1,34 +1,34 @@
...
 
 // styles
-import './Searchbar.css';
+import "./Searchbar.css";
 
 export default function Searchbar() {
-  const [term, setTerm] = useState('')
-  const searchInput = useRef(null)
-  const navigate = useNavigate()
+	const [term, setTerm] = useState("");
```
Comment (neil-borbon) [inline src/components/searchbar/Searchbar.js:8]: Add type to useState

File: src/components/theme-selector/ThemeSelector.css [hunk 6]
```diff
@@ -0,0 +1,31 @@
+.theme-selector {
```
Comment (neil-borbon) [inline src/components/theme-selector/ThemeSelector.css:1]: Use "rem" on all "px" values

File: src/components/theme-selector/ThemeSelector.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+//styles
+import "./ThemeSelector.css";
+
+import modeIcon from "../../assets/mode-icon.svg";
+
+const themeColors = ["#58249c", "#249c6b", "#b70233"];
+
+export default function ThemeSelector() {
+	const { changeColor, changeMode, mode } = useTheme();
+	// const [color, setColor] = useState("#58249c");
```
Comment (neil-borbon) [inline src/components/theme-selector/ThemeSelector.js:13]: remove commented code

File: src/components/theme-selector/ThemeSelector.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
+// import { useState } from "react";
+import { useTheme } from "../../hooks";
+
+//styles
+import "./ThemeSelector.css";
+
+import modeIcon from "../../assets/mode-icon.svg";
+
+const themeColors = ["#58249c", "#249c6b", "#b70233"];
```
Comment (neil-borbon) [inline src/components/theme-selector/ThemeSelector.js:9]: move this to a theme file inside a "constants" folder

File: src/components/theme-selector/ThemeSelector.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+export default function ThemeSelector() {
+	const { changeColor, changeMode, mode } = useTheme();
+	// const [color, setColor] = useState("#58249c");
+
+	const toggleMode = () => {
+		changeMode(mode === "dark" ? "light" : "dark");
```
Comment (neil-borbon) [inline src/components/theme-selector/ThemeSelector.js:16]: Use ?? operator for this ternary

File: src/components/theme-selector/ThemeSelector.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+			</div>
+			<div className='theme-buttons'>
+				{themeColors.map((color) => (
+					<div
+						key={color}
+						onClick={() => changeColor(color)}
+						style={{ background: color }}
+					/>
+				))}
+				{/* <input
```
Comment (neil-borbon) [inline src/components/theme-selector/ThemeSelector.js:37]: Remove commented code

File: src/components/theme-selector/ThemeSelector.js [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+					onClick={toggleMode}
+					src={modeIcon}
+					alt='dark/light toggle icon'
+					style={{ filter: mode === "dark" ? "invert(100%)" : "invert(20%)" }}
+				/>
+			</div>
+			<div className='theme-buttons'>
+				{themeColors.map((color) => (
+					<div
+						key={color}
```
Comment (neil-borbon) [inline src/components/theme-selector/ThemeSelector.js:32]: Key must be unique

File: src/context/ThemeContext.js [hunk 12]
```diff
@@ -6,22 +6,29 @@ const themeReducer = (state, action) => {
 	switch (action.type) {
 		case "CHANGE_COLOR":
 			return { ...state, color: action.payload };
```
Comment (neil-borbon) [inline src/context/ThemeContext.js:8]: Assign value to a variable and return it on the last line of the function, refrain from having multiple return statements in one function

File: src/pages/create/Create.js [hunk 13]
```diff
@@ -1,102 +1,119 @@
...
+	const [method, setMethod] = useState("");
+	const [cookingTime, setCookingTime] = useState("");
+	const [newIngredient, setNewIngredient] = useState("");
+	const [ingredients, setIngredients] = useState([]);
+	const ingredientInput = useRef(null);
+	const navigate = useNavigate();
+	const { color, mode } = useTheme();
 
-  const { postData, data } = useFetch('http://localhost:3000/recipes', 'POST')
+	const { postData, data } = useFetch("http://localhost:3000/recipes", "POST");
```
Comment (neil-borbon) [inline src/pages/create/Create.js:19]: Move the URL to a constants file under "constants" folder

File: src/pages/create/Create.js [hunk 14]
```diff
@@ -1,102 +1,119 @@
...
 
-  const handleSubmit = (e) => {
-    e.preventDefault();
-    console.log(title, method, cookingTime, ingredients)
-    postData({ title, ingredients, method, cookingTime: `${cookingTime} minutes` })
-  }
+	const handleSubmit = (e) => {
+		e.preventDefault();
+		console.log(title, method, cookingTime, ingredients);
```
Comment (neil-borbon) [inline src/pages/create/Create.js:23]: remove console logs

File: src/pages/create/Create.js [hunk 15]
```diff
@@ -1,102 +1,119 @@
...
+					<span>Cooking Time:</span>
+					<input
+						type='number'
+						onChange={(e) => setCookingTime(e.target.value)}
+						placeholder='45'
+						value={cookingTime}
+						required
+					/>
+				</label>
+				<button className='btn' style={{ background: color }}>
```
Comment (neil-borbon) [inline src/pages/create/Create.js:113]: this button has a classname, move the inline style to the classname

File: src/pages/recipe/Recipe.js [hunk 16]
```diff
@@ -1,29 +1,31 @@
...
+import { useParams } from "react-router-dom";
+import { useFetch, useTheme } from "../../hooks/index";
 
 // styles
-import './Recipe.css';
+import "./Recipe.css";
 
 export default function Recipe() {
+	const { id } = useParams();
+	const url = `http://localhost:3000/recipes/${id}`;
```
Comment (neil-borbon) [inline src/pages/recipe/Recipe.js:9]: Move this url to a separate file

File: src/pages/recipe/Recipe.js [hunk 17]
```diff
@@ -1,29 +1,31 @@
...
+		<div className={`recipe ${mode}`}>
+			{error && <p className='error'>{error}</p>}
+			{isPending && <p className='loading'>Loading...</p>}
+			{recipe && (
+				<>
+					<h2 className='page-title'>{recipe.title}</h2>
+					<p className='duration'>Takes {recipe.cookingTime} Cook This</p>
+					<ul>
+						{recipe.ingredients.map((ing) => (
+							<li key={ing}>{ing}</li>
```
Comment (neil-borbon) [inline src/pages/recipe/Recipe.js:23]: Use a better and descriptive variable name for "ing"

File: src/pages/search/Search.js [hunk 18]
```diff
@@ -1,24 +1,25 @@
...
-      {isPending && <p className="loading">Loading...</p>}
-      {data && <RecipeList recipes={data}/>}
-    </div>
-  )
-}
\ No newline at end of file
+	const queryString = useLocation().search;
+	const queryParams = new URLSearchParams(queryString);
+	const query = queryParams.get("q");
+	const url = `http://localhost:3000/recipes?q=${query}`;
```
Comment (neil-borbon) [inline src/pages/search/Search.js:14]: Move URL to separate constants file

**Review by neil-borbon — CHANGES_REQUESTED** [review]
Left a couple of comments, please ping me once the changes are pushed.

File: src/pages/recipe/Recipe.js (hunk 16 above)
Comment (KennethicEnergy) [reply src/pages/recipe/Recipe.js:9]: url is for testing purposes only, will move all data to firebase once next.

File: src/pages/recipe/Recipe.js (hunk 17 above)
Comment (KennethicEnergy) [reply src/pages/recipe/Recipe.js:23]: ing = ingredient

File: src/pages/search/Search.js (hunk 18 above)
Comment (KennethicEnergy) [reply src/pages/search/Search.js:14]: url is for testing purposes only, will move all data to firebase once next.

File: src/components/recipe-list/Recipe-List.css (hunk 2 above)
Comment (KennethicEnergy) [reply src/components/recipe-list/Recipe-List.css:21]: page breaks when converting px to rem values

File: src/components/recipe-list/Recipe-List.css (hunk 3 above)
Comment (KennethicEnergy) [reply src/components/recipe-list/Recipe-List.css:64]: page breaks when converting px to rem values

File: src/components/searchbar/Searchbar.js (hunk 5 above)
Comment (KennethicEnergy) [reply src/components/searchbar/Searchbar.js:8]: app currently doesn't support typescript.

File: src/components/theme-selector/ThemeSelector.css (hunk 6 above)
Comment (KennethicEnergy) [reply src/components/theme-selector/ThemeSelector.css:1]: page breaks when converting px to rem values

File: src/pages/create/Create.js (hunk 15 above)
Comment (KennethicEnergy) [reply src/pages/create/Create.js:113]: currently using css for styling and inline style is for themeing.

**Pull request merged**